instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -183,11 +189,16 @@ def process_request(self, request):
             carrier_getter = wsgi_getter
             collect_request_attributes = wsgi_collect_request_attributes
 
-        token = attach(extract(carrier, getter=carrier_getter))
-
+        token = context = None
+        span_kind = SpanKind.INTERNAL
+        if get_current_span() is INVALID_SPAN:
+            context = extract(request_meta, getter=carrier_getter)
+            token = attach(context)
+            span_kind = SpanKind.SERVER
         span = self._tracer.start_span(
             self._get_span_name(request),
-            kind=SpanKind.SERVER,
+            context,
+            kind=span_kind,
```

Comment:
@owais do we always want the span kind to be INTERNAL when some other span already exists or do we want it only when there is a SERVER span?

Reply:
I think if any span exists that we use as a parent, we should use INTERNAL as SERVER span must have a CLIENT span as the parent IMO